kiali.yaml

```
@@ -66,15 +66,117 @@ subpackages:
 
 test:
   pipeline:
-    - name: Verify Kiali API
+    - name: Verify Kiali API Binary
       runs: |
         # Check if binary runs
         /opt/kiali/kiali --help
-    - name: Verify Kiali UI
+    - name: Verify Kiali UI Assets
       runs: |
-        # Make sure webroot HTML index file exists
+        # Verify webroot HTML index file exists and contains expected content
+        test -f /opt/kiali/console/index.html
         grep '</html>' /opt/kiali/console/index.html
 
+        # Verify static assets presence using find instead of tree
+        find /opt/kiali/console -type f -name "*.js" | grep -q "static/js" || exit 1
+        find /opt/kiali/console -type f -name "*.css" | grep -q "static/css" || exit 1
+
+        # Verify essential UI files exist
+        test -f /opt/kiali/console/manifest.json
+        test -f /opt/kiali/console/kiali_icon_lightbkg_16px.png
+    - uses: test/kwok/cluster
+    - name: Verify Kiali Health Endpoint
+      runs: |
+        # Get kubeconfig from Kwok cluster and write to file
+        kwokctl get kubeconfig > kubeconfig.yaml
+        if [ ! -s kubeconfig.yaml ]; then
+          echo "Error: Failed to get kubeconfig from Kwok cluster"
+          exit 1
+        fi
+
+        if ! kubectl --kubeconfig kubeconfig.yaml get nodes; then
+          echo "Error: Invalid kubeconfig"
+          cat kubeconfig.yaml
+          exit 1
+        fi
+
+        # Extract server URL from kubeconfig
+        SERVER_URL=$(kubectl --kubeconfig kubeconfig.yaml config view -o jsonpath='{.clusters[0].cluster.server}')
+        if [ -z "$SERVER_URL" ]; then
+          echo "Error: Could not extract server URL from kubeconfig"
+          exit 1
+        fi
```

Let's make this less verbose, `kubectl` already `exit`s with `1` on error anyway.
```suggestion
        kwokctl get kubeconfig > kubeconfig.yaml
        kubectl --kubeconfig kubeconfig.yaml get nodes
        SERVER_URL=$(kubectl --kubeconfig kubeconfig.yaml config view -o jsonpath='{.clusters[0].cluster.server}')
```